src/lib.rs

```diff
@@ -25,22 +31,30 @@
 //!
 //! # Example
 //!
-//! This example will show how to print "Hello, world!" on the host.
+//! ## Using `hio::HStdout`
 //!
-//! Target program:
+//! This example will demonstrate how to print formatted strings.
 //!
-//! ```
+//! ```rust
 //! #[macro_use]
 //! extern crate cortex_m_semihosting;
 //!
+//! use cortex_m_semihosting::hio;
+//! use core::fmt::Write;
+//!
 //! // This function will be called by the application
-//! fn print() {
-//!     // File descriptor (on the host)
-//!     const STDOUT: usize = 1; // NOTE the host stdout may not always be fd 1
-//!     static MSG: &'static [u8] = b"Hello, world!\n";
+//! fn print() -> Result<(), core::fmt::Error> {
+//!     let mut stdout = match hio::hstdout() {
+//!         Ok(fd) => fd,
+//!         Err(()) => return Err(core::fmt::Error),
+//!     };
 //!
-//!     // Signature: fn write(fd: usize, ptr: *const u8, len: usize) -> usize
-//!     let r = unsafe { syscall!(WRITE, STDOUT, MSG.as_ptr(), MSG.len()) };
+//!     let language = "Rust";
+//!     let ranking = 1;
+//!
+//!     stdout.write_fmt(format_args!("{} on embedded is #{}!", language, ranking))?;
```

Comment:
It's more rustic to write this as `writeln!(stdout, "{} on embedded is #{}!", language, ranking)?`.